openlibrary/macros/BestbookModal.html

```diff
@@ -0,0 +1,79 @@
+$def with (work, link_markup, id, classes='', reload_id=None, edition_key=None)
+$# :param Work work: The work that is being reviewed
+$# :param str link_markup: arkup for element that triggers observations modal
+$# :param str id: Unique identifier for this modal
+$# :param str classes: HTML classes for this component
+$# :param str reload_id: Reference to UI component that will update when observations are added or removed
+
+$ topics = work.get_subjects()
+$ username = ctx.user and ctx.user.key.split('/')[-1]
+
+$# The following data is accessed via JS:
+$ context = {
+  $ "username": username,
+  $ "work": work.key,
+  $ "id": id
+  $ }
+
+$# Render the subject input field
+$jsdef render_subject_field(name, data):
+    <div class="input">
+        $code:
+            subjects = []
+            for s in data:
+                subjects.append('"' + s + '"' if ',' in s else s)
+            subject_str = ', '.join(subjects)
+            rows = len(subject_str) // 95 + 1
+        <textarea id="about-$name" name="work--$name" rows="$rows" wrap="soft" tabindex="0">$subject_str</textarea>
+    </div>
+
+$jsdef render_subject_autocomplete_item(item):
+    <div class="ac_subject" title="$_('Select this subject')">
+        <span class="name">$item.name</span>
+    </div>
+
+$if reload_id:
+  $ context['reloadId'] = reload_id
+
+$ work_id = work.key.split('/')[2]
+$ cover_url = "https://covers.openlibrary.org/b/id/{}-M.jpg".format(work_id[2:-1])
+
+$ src = cover_url if work_id else '/images/icons/avatar_book.png'
+
+$if any(work.get_authors()):
+  $ author_names = ", ".join(a.get("name", "Name missing") for a in work.get_authors())
+
+<div class="$classes">
+  $:link_markup
+  $if ctx.user:
+    <div class="hidden">
+      <div id="$id-metadata-form" class="floaterAdd metadata-form" data-context="$json_encode(context)">
+        <div class="floaterHead">
+          <h2>$_("Best Book Award")</h2>
+          <a class="dialog--close">&times;<span class="shift">$_("Close")</span></a>
+        </div>
+        <div class="bestbook_body">
+          <div class="bestbook_data">
+            <h3>$:work.title</h3>
```

Comment:
Careful using `$:` here as it will render the work.title as html as opposed to just inserting its text and this could cause js xss injection!
```suggestion
            <h3>$work.title</h3>
```